src/trackers.js

```diff
@@ -1,227 +1,255 @@
         }
 
-        let trackerFromList = this.checkTrackerLists(urlSplit, currLocation, urlToCheck, requestType)
-        if (trackerFromList) {
-            let commonParent = this.getCommonParentEntity(currLocation, urlToCheck)
-            if (commonParent) {
-                return this.addCommonParent(trackerFromList, commonParent)
-            }
-            return trackerFromList
+        // tracker object returned by getTrackerData
+        const tracker = {
+            definition: null,
+            action: false,
+            redirectUrl: null,
+            owner: null,
+            firstParty: false,
+            matchedRule: null,
+            matchedRuleException: false
         }
 
-        // embedded tweet option
-        // a more robust test for tweet code may need to be used besides just
-        // blocking platform.twitter.com
-        if (ops.embeddedTweetsEnabled === false &&
-                /platform.twitter.com/.test(urlToCheck)) {
-            let tracker = { parentCompany: 'Twitter', url: 'platform.twitter.com', type: 'Analytics', block: true }
-            let commonParent = this.getCommonParentEntity(currLocation, urlToCheck)
+        // finds a tracker definition by iterating over the whole trackerList and finding the matching tracker.
+        tracker.definition = this.findTracker(tracker, requestData)
 
-            if (commonParent) {
-                return this.addCommonParent(tracker, commonParent)
-            }
-            return tracker
+        if (!tracker.definition) {
+            return null
         }
 
-        return false
-    }
+        // finds a matching rule by iterating over the rules in tracker.data and sets redirectUrl.
+        tracker.matchedRule = this.findRule(tracker, requestData)
 
-    // add common parent info to the final tracker object returned by isTracker
-    addCommonParent (trackerObj, parentName) {
-        trackerObj.parentCompany = parentName
-        trackerObj.block = false
-        trackerObj.reason = 'first party'
-        return trackerObj
-    }
+        // sets tracker.exception by looking at tracker.rule exceptions (if any)
+        tracker.matchedRuleException = this.matchesException(tracker, requestData)
 
-    checkTrackerLists (urlSplit, currLocation, urlToCheck, requestType) {
-        // Look up trackers by parent company. This function also checks to see if the poential
-        // tracker is related to the current site. If this is the case we consider it to be the
-        // same as a first party requrest and return
-        let trackerByParentCompany = this.checkTrackersWithParentCompany(urlSplit, currLocation)
-        if (trackerByParentCompany) {
-            return trackerByParentCompany
-        }
+        // compare the site owner to the tracker owner and set firstParty
+        tracker.firstParty = this.setFirstParty(tracker, requestData)
+
+        this.setAction(tracker, requestData)
+
+        return tracker
```

Comment:
after looking at this a bit longer I do find it a bit hard to follow that some of those methods are modifying `tracker` (often in unexpected ways) and some return a value that gets assigned to tracker property. I'd change that to be sth like this:

```js
const tracker = this.findTracker(request);

if(!tracker) {
  return null;
}

const matchedRule = this.findMatchingRule(tracker.rules, request);

if (matchedRule && matchedRule.surrogate) {
   // getSurrogateURL could find given surrogate and add the 'data:' prefix
   const redirectUrl = this.getSurrogateURL(matchedRule.surrogate);
}

// we should be able to reuse `findRule` for exceptions
const matchedRuleException = this.findMatchingRule(tracker.exceptions, request);

const trackerOwner = this.findTrackerOwner(tracker);

const websiteOwner = this.findWebsiteOwner(request);

const firstParty = trackerOwner === websiteOwner;

// only pass info that it needs
const {action, reason} = this.shouldBlock({
  firstParty,
  matchedRule,
  matchedRuleException,
  defaultAction: tracker.defaultAction,
  redirectUrl
});

// only return info that we need 
return {
  action,
  reason,
  redirectUrl,
  …
};
```